pgsql_big_dedupe_example/pgsql_big_dedupe_example.py

```diff
@@ -220,34 +261,80 @@ def cluster_ids(clustered_dupes):
             cur.execute("CREATE TABLE blocking_map "
                         "(block_key text, donor_id INTEGER)")
 
+    # Compute `(block_key, donor_id)` tuples and write on `blocking_map` in parallel,
+    # but only for the non-index predicates. Only those can run in parallel
+    # because they don't need global predicate indexes.
+    # We cannot share predicate indexes across Python processes because it would consume
+    # too much RAM
+    print('computing blocking map in parallel (non-index predicates)')
+
+    predicates_without_index = []
+    predicates_with_index = []
+    for full_predicate in deduper.fingerprinter.predicates:
+        if any(predicate in deduper.fingerprinter.index_predicates
+               for predicate in full_predicate):
+            predicates_with_index.append(full_predicate)
+        else:
+            predicates_without_index.append(full_predicate)
+
+    # Use only predicates WITHOUT indexes for parallel blocking
+    deduper.fingerprinter.predicates = predicates_without_index
```

Comment:
I've found a problem here... due to these lines:

https://github.com/dedupeio/dedupe/blob/c250911590a72e77612bf9549c78c90e2fb01705/dedupe/blocking.py#L89-L91

The `enumerate` is causing the predicate to have a different `pred_id` compared to the serial version. This only happens if there's an index predicate. What would be the best way to solve this? Changing Dedupe internally or doing some workaround here?

Reply:
hmm.. that is interesting. enumerate is just identifying the predicate. we could accomplish the same thing by having the predicates be hashable and using the hash.

Reply:
that looks good to me.